catalyst-gateway/bin/src/service/api/documents/put_document/unprocessable_content_request.rs

```diff
@@ -1,31 +1,38 @@
 //! Bad Document PUT request.
 
-use poem_openapi::{types::Example, Object};
+use poem_openapi::{
+    types::{Example, ParseFromJSON},
+    Object,
+};
+
+use crate::service::common;
 
 /// Put Document Validation Error.
-#[derive(Object, Default)]
+#[derive(Object)]
 #[oai(example = true)]
 pub(crate) struct PutDocumentUnprocessableContent {
     /// Error messages.
-    #[oai(validator(max_length = "100", pattern = "^[0-9a-zA-Z].*$"))]
-    error: String,
+    error: common::types::generic::error_msg::ErrorMessage,
     /// Error report JSON object.
     #[oai(skip_serializing_if_is_none)]
-    report: Option<serde_json::Value>,
+    report: Option<common::objects::generic::json_object::JSONObject>,
 }
 
 impl PutDocumentUnprocessableContent {
     /// Create a new instance of `ConfigBadRequest`.
     pub(crate) fn new(error: &(impl ToString + ?Sized), report: Option<serde_json::Value>) -> Self {
         Self {
-            error: error.to_string(),
-            report,
+            error: error.to_string().into(),
+            report: common::objects::generic::json_object::JSONObject::parse_from_json(report).ok(),
```

Comment:
Better to add a `From<serde_json::Value>` trait implementation for the `common::objects::generic::json_object::JSONObject` and in this place make something like this
```rust
report : report.map(Into::into)
```